### DISCUSSION TITLE
Does Mesa support changing agent sizes in a continuous space?

### DISCUSSION THREAD

**Discussion by Lee-Talman** [mesa/mesa#1829] [Q&A] [answered]
I'm modeling a 2D bacterial culture where the individual bacteria (agents) are initially small, circular, and tightly-packed. As they grow, they will become more cylindrical (i.e. grow faster in the Y direction than their X direction) and "push" another around. 

Repast can theoretically(?) do this, but I have a much stronger grasp of Python than Java and I would like to use Mesa for this task. Does anyone know if this is possible?

**Comment by rht** [comment]
There shouldn't be any limitation in modelling the boundaries of the agents. You could do a check whether the bacteria are in collision with each other on the continuous space. You could also visualize this via Matplotlib scatter plot with the new Python-based Solara frontend that we have just added in Mesa 2.x.

**Reply by rht** [reply, marked as answer]
#1830 implements drawing the continuous space. You can try this with the boid flockers example https://github.com/projectmesa/mesa-examples/tree/main/examples/boid_flockers, 
```python
from mesa.experimental import JupyterViz

from boid_flockers.model import BoidFlockers


def boid_draw(agent):
    return {"color": "tab:red"}


model_params = {
    "population": 100,
    "width": 100,
    "height": 100,
    "speed": 5,
    "vision": 10,
    "separation": 2,
}

page = JupyterViz(
    BoidFlockers,
    model_params,
    measures=[],
    name="BoidFlockers",
    agent_portrayal=boid_draw,
)
page  # noqa
```

**Reply by Lee-Talman** [reply]
This worked perfectly, thank you!
